src/main/webapp/resources/js/pages/projects/create/CreateProjectLayout.js

```diff
@@ -0,0 +1,166 @@
+import { Button, Col, Form, Modal, notification, Row, Steps } from "antd";
+import React from "react";
+import { createProject } from "../../../apis/projects/create";
+import { setBaseUrl } from "../../../utilities/url-utilities";
+import { CreateProjectDetails } from "./CreateProjectDetails";
+import { CreateProjectSamples } from "./CreateProjectSamples";
+
+/**
+ * React component to handle the layout of the Create New Project and
+ * display a modal to contain it.
+ *
+ * @param  {JSX.Element} children - A clickable element used to open the modal
+ * @returns {JSX.Element}
+ * @constructor
+ */
+export function CreateProjectLayout({ children }) {
+  const [form] = Form.useForm();
+  const [visible, setVisible] = React.useState(false);
+  const [current, setCurrent] = React.useState(0);
+  const [loading, setLoading] = React.useState(false);
+
+  const steps = [
+    {
+      title: i18n("CreateProjectLayout.details"),
+      content: <CreateProjectDetails form={form} />,
+    },
+    {
+      title: i18n("CreateProjectLayout.samples"),
+      content: <CreateProjectSamples form={form} />,
+    },
+  ];
+
+  const validateMessages = {
+    required: i18n("CreateProjectDetails.required"),
+    string: {
+      min: i18n("CreateProjectDetails.length"),
+    },
+    types: {
+      url: i18n("CreateProjectDetails.url"),
+    },
+  };
+
+  /**
+   * Once the form is filled out, this is the submit to server call.
+   * After a successful call, the user will be redirected to the new project.
+   */
+  const submit = () => {
+    setLoading(true);
+    form
+      .validateFields()
+      .then((values) =>
+        createProject(values).then(
+          ({ id }) => (window.location.href = setBaseUrl(`projects/${id}`))
```

Comment:
Could we not use navigate from the reach router package here?

Reply:
No because it really is taking you to a new entry, not just another path.